devicetypes/smartthings/zigbee-battery-accessory-dimmer.src/zigbee-battery-accessory-dimmer.groovy

```diff
@@ -0,0 +1,157 @@
+		capability "Actuator"
+		capability "Battery"
+		capability "Configuration"
+		capability "Health Check"
+		capability "Switch"
+		capability "Switch Level"
+		
+		fingerprint profileId: "0104", inClusters: "0000,0001,0003,0020,FC11", outClusters: "0003,0004,0006,0008,FC10", manufacturer: "sengled", model: "E1E-G7F", deviceJoinName: "Sengled Smart Switch"
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"switch", type: "lighting", width: 6, height: 4, canChangeIcon: true){
+			tileAttribute ("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "on", label:'${name}', action:"switch.off", icon:"st.switches.light.on", backgroundColor:"#00A0DC", nextState:"turningOff"
+				attributeState "off", label:'${name}', action:"switch.on", icon:"st.switches.light.off", backgroundColor:"#ffffff", nextState:"turningOn"
+				attributeState "turningOn", label:'${name}', action:"switch.off", icon:"st.switches.light.on", backgroundColor:"#00A0DC", nextState:"turningOff"
+				attributeState "turningOff", label:'${name}', action:"switch.on", icon:"st.switches.light.off", backgroundColor:"#ffffff", nextState:"turningOn"
+			}
+			tileAttribute ("device.level", key: "SLIDER_CONTROL") {
+				attributeState "level", action:"switch level.setLevel"
+			}
+			tileAttribute ("device.battery", key: "SECONDARY_CONTROL") {
+				attributeState "battery", label: 'battery ${currentValue}%', unit: "%"
+			}
+		}
+		main "switch"
+		details(["switch"])
+	}
+}
+
+def getDOUBLE_STEP() {10}
+def getSTEP() {5}
+
+def parse(String description) {
+
+	def descMap = zigbee.parseDescriptionAsMap(description)
+	if (isSengledSwitch()) {
+		handleSengledSwitchEvents(descMap)
+	}
+	handleBatteryEvents(descMap)
+}
+
+private boolean isSengledSwitch() {
+	device.getDataValue("model") == "E1E-G7F"
+}
+
+def handleSengledSwitchEvents(descMap) {
+	if (descMap?.clusterInt == 0xFC10) {
+		def currentLevel = device.currentValue("level") as Integer ?: 0
+		def value = 0
+
+		switch(descMap.data[0]) {
+			case '01':
+				//short press of 'ON' button
+				sendEvent(name: "switch", value: "on")
+				break
+			case '02':
+				// move up
+				if (descMap.data[2] == '02') {
+					//long press of 'BRIGHTEN' button
+					value = Math.min(currentLevel + DOUBLE_STEP, 100)
+				} else if (descMap.data[2] == '01') {
+					//short press of 'BRIGHTEN' button
+					value = Math.min(currentLevel + STEP, 100)
+				}
+				sendEvent(name: "switch", value: "on")
+				sendEvent(name: "level", value: value)
+				break
+			case '03':
+				//move down
+				if (descMap.data[2] == '02') {
+					//long press of 'DIM' button
+					value = Math.max(currentLevel - DOUBLE_STEP, 0)
+				} else if (descMap.data[2] == '01') {
+					//short press of 'DIM' button
+					value = Math.max(currentLevel - STEP, 0)
+				}
+
+				if (value == 0) {
+					sendEvent(name: "switch", value: "off")
+				}
+				sendEvent(name: "level", value: value)
```

Comment:
Should `level` only be set if value != 0? That's what we usually do for dimmers

Reply:
yeah I think Tom is right here

Reply:
Yeah, can be, but I think that version that sends 0 value and 'off' event is slighly better. We have step 5 or 10, so then the current value on slider is 5, when button is pushed, we will send 'off' event, but value 5 will still be set. In this edge condition I would send event with value 0 or 1, to make sure that the lowest level was achieved. But I also feel comfortable with standard approach.